src/app/services/myinfo/myinfo.service.ts

```diff
@@ -0,0 +1,301 @@
+  }) {
+    this.#spCookieMaxAge = spCookieMaxAge
+
+    const { myInfoClientMode, myInfoKeyPath } = myInfoConfig
+    let myInfoGovClient: MyInfoGovClient
+    const myInfoPrefix =
+      myInfoClientMode === MyInfoClientMode.Staging
+        ? MYINFO_STG_PREFIX
+        : MYINFO_PROD_PREFIX
+    if (nodeEnv === Environment.Prod) {
+      myInfoGovClient = new MyInfoGovClient({
+        realm,
+        singpassEserviceId,
+        privateKey: fs.readFileSync(myInfoKeyPath),
+        appId: myInfoPrefix + singpassEserviceId,
+        mode: myInfoClientMode,
+      })
+    } else {
+      myInfoGovClient = new MyInfoGovClient({
+        realm,
+        singpassEserviceId,
+        privateKey: fs.readFileSync(myInfoKeyPath),
+        appId: myInfoPrefix + singpassEserviceId,
+        mode: MyInfoClientMode.Dev,
+      })
+      myInfoGovClient.baseUrl = MYINFO_DEV_BASE_URL
+    }
+
+    this.#myInfoClientBreaker = new CircuitBreaker(
+      (params) => myInfoGovClient.getPersonBasic(params),
+      {
+        errorThresholdPercentage: 80, // % of errors before breaker trips
+        timeout: 5000, // max time before individual request fails, ms
+        rollingCountTimeout: 30000, // width of statistical window, ms
+        volumeThreshold: 5, // min number of requests within statistical window before breaker trips
+      },
+    )
+  }
+
+  /**
+   * Fetches MyInfo person detail with given params.
+   * This function has circuit breaking built into it, and will throw an error
+   * if any recent usages of this function returned an error.
+   * @param params The params required to retrieve the data.
+   * @param params.uinFin The uin/fin of the person's data to retrieve.
+   * @param params.requestedAttributes The requested attributes to fetch.
+   * @param params.singpassEserviceId The eservice id of the form requesting the data.
+   * @returns the person object retrieved.
+   * @throws an error on fetch failure or if circuit breaker is in the opened state. Use {@link CircuitBreaker#isOurError} to determine if a rejection was a result of the circuit breaker or the action.
+   */
+  fetchMyInfoPersonData(
+    params: IPersonBasicRequest,
+  ): ResultAsync<IPersonBasic, CircuitBreakerError | FetchMyInfoError> {
+    return ResultAsync.fromPromise(
+      this.#myInfoClientBreaker.fire(params),
+      (error) => {
+        const isCircuitOpen = CircuitBreaker.isOurError(error)
+        const logMessage = isCircuitOpen
+          ? 'Circuit breaker tripped'
+          : 'Error retrieving from MyInfo'
+        logger.error({
+          message: logMessage,
+          meta: {
+            action: 'fetchMyInfoPersonData',
+            meta: {
+              requestedAttributes: params.requestedAttributes,
+              eServiceId: params.singpassEserviceId,
+            },
+          },
+          error,
+        })
+        return isCircuitOpen
+          ? new CircuitBreakerError(logMessage)
+          : new FetchMyInfoError(logMessage)
```

Comment:
it'll be simpler to use an if-else statement here instead of having to perform ternary op on `isCircuitOpen` twice.